### PR TITLE
DX: Reduce usage of PhpCsFixer\DocBlock\Line::__toString

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

File: src/DocBlock/Line.php [hunk 1]
```diff
@@ -42,6 +42,8 @@ public function __construct($content)
      * Get the string representation of object.
      *
      * @return string
+     *
+     * @deprecated will be removed in 3.0, use getContent
```
Comment (julienfalque) [inline src/DocBlock/Line.php]: Shouldn't the method make a `@trigger_error('...', E_USER_DEPRECATION)` call?

Comment (SpacePossum) [reply src/DocBlock/Line.php]: risky, because a lot of IDE's call this method to show preview data when debugging

Comment (kubawerlos) [reply src/DocBlock/Line.php]: Wouldn't `@` make it transparent for IDEs?

**Comment by kubawerlos** [comment]
Anyway, the magic is heavily used - in `DocBlock` and `Annotation` classes as well - should we get rid of all the `__toString` functions and replace:

```php
implode('', $lines);
```
with:

```php
implode(
    '',
    array_map(
        function (Line $line) {
            return $line->getContent();
        },
        $lines
    )
);
```
?

**Comment by kubawerlos** [comment]
> Please, provide reasoning why to deprecate this in first place.

@keradus not sure not now, initially was to simply avoid magic method.

**Comment by keradus** [comment]
1. no reasoning is indicator to not pursuit the idea
2. lack of `@trigger_error` is an issue, especially, as you noticed yourself, we reduce usage of that magic method, but not get rid of it
3. I removed `deprecation` part of this PR and now, I will merge it as is - if we want to deprecate sth indeed, we need to have good reasoning for it, deprecation notification in place and stop using deprecated code in our codebase.

**Comment by keradus** [comment]
Already applied in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4088

**Comment by keradus** [comment]
note:
i'm not for using magic methods everywhere, but we may use them sometimes.

If you want to get rid of them, please deprecate them (with real deprecation notice) everywhere, in all `PhpCsFixer\DocBlock\*` and `AllowedValueSubset` (the `__invoke`)

**Comment by SpacePossum** [comment]
>1. no reasoning is indicator to not pursuit the idea

I would like to note that using `__toString()` comes with some drawbacks which is why I never use it:
1 - finding "usages" is near impossible, because;
2 - it allows for implicit string casting of an object
3 - it is not a descriptive method name
4 - it is invalid to throw exceptions from the method, which makes it more limited than any regular method for no reason

Magic methods like `__invoke` and `__isset` have a far better defined role and definition and besides being magic have nothing in common with `__toString`.

I'm fine with not merging though as it might not be worth adding this BC break now :)

**Comment by keradus** [comment]
> > not sure not now, initially was to simply avoid magic method.

> no reasoning is indicator to not pursuit the idea

just saying that change for a change is not a reason, it has to have background

as said later, i'm not negative for such change, but it has to be applied fully

**Pull request closed without merging**